cli/kubernetes/installer.go

```diff
@@ -54,15 +59,49 @@ type OptionsReader interface {
 // InteractiveOptionsReader which will ask in the terminal.
 // This method only populates what is possible and leaves the rest empty.
 // TODO: Implement another method to validate that all options have been set.
-func (i *Installer) PopulateNeededOptions(reader OptionsReader) error {
+func (i *Installer) PopulateNeededOptions(cmd *cobra.Command, reader OptionsReader) error {
 	if reader == nil {
 		reader = NewInteractiveOptionsReader(os.Stdout, os.Stdin)
 	}
 
 	var err error
 	newOptions := InstallationOptions{}
 	for _, opt := range i.NeededOptions {
-		opt.Value, err = reader.Read(opt)
+		var skipInteractive = false
```

Comment:
The original idea  of rthe OptionsReader was that you could make multiple calls to `PopulateNeededOptions` with a different OptionsReader each time. E.g.

```
...
PopulateNeededOptions(cobraReader)
PopulateNeededOptions(yamlReader)
PopulateNeededOptions(interactiveReader)
...
```
each call would fill in as many options as possible. This way you can have a set of options filled in  from command line, another set from a yaml file and some other interactively. This may be useful when for example the user stores most values in a yaml file but then wants to overwrite some on the command line. Secrets should probably never be written on disk thus a final interactive pass may be needed. (e.g. for the admin password)

With the above in mind, maybe we can wrap the cobra logic in a separate OptionsReader. What do you think?

Reply:
That sounds good. The main thing I see right now is the need for some kind flag indicating that the `opt.Value` is valid. That way each reader can skip the options already set by previous readers.

 Will look into this on Monday.

Reply:
Done with commits 118083a58150f52e11e39395b8b1f81a6a94d66d, ceeb8e1be9dc4c70db57e8318037dd65f78d20d3, and eb6d311ad65607705e12870fdcfdb76aab4be329